vim-patch: runtime updates

justinmk opened the pull request:
```
Found errors in Test_filetype_detection():
function RunTheTest[35]..Test_filetype_detection[2]..CheckItems line 9: with file name: file.lpc: Expected 'pike' but got 'lpc'
function RunTheTest[35]..Test_filetype_detection[2]..CheckItems line 9: with file name: file.ulpc: Expected 'pike' but got 'lpc'
```

janlazo committed:
vim-patch:8.0.1438: filetype detection test not updated for change

Problem:    Filetype detection test not updated for change.
Solution:   Update the test.
https://github.com/vim/vim/commit/0479e910c423d71e2b96bc721feffad5808e767a

janlazo committed:
vim-patch:8.0.1457: clojure now supports a shebang line

Problem:    Clojure now supports a shebang line.
Solution:   Detect clojure script from the shebang line. (David Burgin,
            closes vim/vim#2570)
https://github.com/vim/vim/commit/8fd2ffc530f8f01afac53903e6ff95867afd94d7

janlazo committed:
vim-patch:8.0.1458: filetype detection test does not check all scripts

Problem:    Filetype detection test does not check all scripts.
Solution:   Add most scripts to the test
https://github.com/vim/vim/commit/ddb349369d107c14fad9c38baf2f0e2b8514fbf0

janlazo committed:
vim-patch:8.0.1465: python2 and python3 detection not tested

Problem:    Python2 and python3 detection not tested. (Matej Cepl)
Solution:   Add test for detecting python2 and python3.  Also detect a script
            using "js" as javascript.
https://github.com/vim/vim/commit/4bc0bed53695ac67db8d601f2a15e48e7a196688

janlazo committed:
vim-patch:8.1.0484: some file types are not recognized

Problem:    Some file types are not recognized.
Solution:   Update the file type detection.
https://github.com/vim/vim/commit/38654503b04dd5ff4813f81892d9f62db1ff01b9

janlazo committed:
vim-patch:8.1.0498: /etc/gitconfig not recognized at a gitconfig file

Problem:    /etc/gitconfig not recognized at a gitconfig file.
Solution:   Add pattern to filetype detection. (closes vim/vim#3568)
https://github.com/vim/vim/commit/d474686a09ef3f7529e65abe00cf8cd2ea8a95eb

justinmk committed:
vim-patch:a2a80162deb1

Update runtime files.
https://github.com/vim/vim/commit/a2a80162deb1e96e16b097dfe48b61b6eb0824bf

justinmk committed:
vim-patch:461fe50fea24

patch 8.0.1367: terminal test hangs, executing abcde

Problem:    terminal test hangs, executing abcde. (Stucki)
Solution:   Rename abcde to abxde.
https://github.com/vim/vim/commit/461fe50fea245b2b199d92ebce4d9875d856bd27

justinmk committed:
vim-patch:4697399e8c80

move netrw back to the previous version
https://github.com/vim/vim/commit/4697399e8c805325009351a2488e3da530d0af38

justinmk committed:
vim-patch:f0b03c4e98f8

Update runtime files
https://github.com/vim/vim/commit/f0b03c4e98f8a7184d8b4a5d702cbcd602426923

Note: haskell changes were included in 942f3587c38a83cf6486a0b779765b54a1648493

justinmk committed:
vim-patch:df980db69b83

update a few runtime files
https://github.com/vim/vim/commit/df980db69b831381c37c3e2973a6eefa10df305c

Note: omit hl-Terminal, Nvim has its own way of highlighting :terminal.

justinmk committed:
vim-patch:40962ec9c0e7

Update runtime files.
https://github.com/vim/vim/commit/40962ec9c0e7b8699e101182b06ddd39dc0e1212

justinmk committed:
vim-patch:7254067ee970

Update runtime files.
https://github.com/vim/vim/commit/7254067ee970686cc3ff4a1effc3e49e9192a5c1

justinmk committed:
vim-patch:22f1d0e35eb0

Updated runtime files.

Add Serbian translations and spell checking.
https://github.com/vim/vim/commit/22f1d0e35eb0e503c2dc7813a973c7e6ca1b7dde

janlazo commented:
Does it pass after including vim-patch:8.0.1438? I can rebase #9015 on top of this PR if this PR is strictly for `runtime/` patches.

justinmk committed:
Merge #9015

justinmk committed:
vim-patch:b5b7562475ad

Update runtime files.
https://github.com/vim/vim/commit/b5b7562475ad032a174b893286172de0d2c157cd

---

NA patch (Nvim does not ship with spell files):
vim-patch:3ad8772ef02e
Include Serbian spell input files
https://github.com/vim/vim/commit/3ad8772ef02e4352b0fa4e2697abc357734b62e1

justinmk committed:
vim-patch:98ef233e14fa

Update runtime files. Convert a couple of help files to utf-8.
https://github.com/vim/vim/commit/98ef233e14faaaf42e0342da1803d493ce0337b2

justinmk committed:
vim-patch:ab943431d8fc

Update runtime files
https://github.com/vim/vim/commit/ab943431d8fcd856008a025b0e5652dd4b8007fc

justinmk committed:
vim-patch:51ad4eaa22e1

Update runtime files
https://github.com/vim/vim/commit/51ad4eaa22e15cf1fe6c45d82c7e1371e00401a4

---

NA:
vim-patch:6176697203b0

justinmk committed:
vim-patch:7dda86f2ff35

Update runtime files.
https://github.com/vim/vim/commit/7dda86f2ff35bb80afce4da24782fd58216bbe50

justinmk committed:
vim-patch:15142e27aaaf

Update runtime files and translations
https://github.com/vim/vim/commit/15142e27aaafa15b72d1042c25fbb5e4f12b6736

justinmk committed:
vim-patch:85eee130f44a

Update runtime files.
https://github.com/vim/vim/commit/85eee130f44a2201d88ca2aeff0af3b11dd75fa9

justinmk committed:
vim-patch:7db25fed5de1

Update runtime files.
https://github.com/vim/vim/commit/7db25fed5de1be922b8cbb0328149469606a0424

justinmk committed:
vim-patch:eb3dc87f0139

Updated runtime files
https://github.com/vim/vim/commit/eb3dc87f01391bb075d97aef3d00f91b4e08a25c

justinmk committed:
vim-patch:7c63fbc46e21

Updated runtime files.
https://github.com/vim/vim/commit/7c63fbc46e218cad266641a743a6c1353c2f00d4

justinmk committed:
vim-patch:2f01889055f1

Runtime file and translation updates.
https://github.com/vim/vim/commit/2f01889055f14842c1d4f9abd91139fc6bd95f15

justinmk committed:
vim-patch:0b0f0992d46e

Update runtime files.
https://github.com/vim/vim/commit/0b0f0992d46ef02fdfc1240744bef91d4299c1df

justinmk committed:
vim-patch:d2f3a8b87873

Update runtime files.
https://github.com/vim/vim/commit/d2f3a8b8787333abf2300d38836b196955f10c00

justinmk committed:
vim-patch:675e8d6adb88

Update runtime files.
https://github.com/vim/vim/commit/675e8d6adb88d5b38087870dfda51c0205070741

---

NA:
vim-patch:ffd112edc6a8
Fix missing patch number.
https://github.com/vim/vim/commit/ffd112edc6a8f72b66c06207bff4431f6aec7b6f

justinmk committed:
vim-patch:6dc819b1299e

Updated runtime and language files.
https://github.com/vim/vim/commit/6dc819b1299e1d9f99303568772ade544d5c1322

justinmk committed:
vim-patch:fd35811ca528

Update runtime files, add Danish translations.
https://github.com/vim/vim/commit/fd35811ca528de7612f9571bce20ef4c392fe5f7

justinmk committed:
vim-patch:b477af2260d9

Update runtime files.
https://github.com/vim/vim/commit/b477af2260d9bc7ae7f743f0a14265d7ee12ee09

justinmk committed:
vim-patch:a9604e614517

Update runtime files.
https://github.com/vim/vim/commit/a9604e61451707b38fdcb088fbfaeea2b922fef6

justinmk committed:
vim-patch:91f84f6e11cd

Update runtime files.
https://github.com/vim/vim/commit/91f84f6e11cd879d43d651c0903d85bff95f0716

justinmk committed:
vim-patch:d2855f5454c5

Update runtime files.
https://github.com/vim/vim/commit/d2855f5454c5c6c5f786b228c5b67757edfefcb1

justinmk committed:
vim-patch:d473c8c10126

Update runtime files.
https://github.com/vim/vim/commit/d473c8c101262702ea9eeb14907ee20a786942b2

justinmk commented:
@janlazo I merged #9015 into this.

justinmk committed:
runtime/optwin.vim: feature-check 'pw' option

Option not implemented yet.
